app/scripts/modules/core/src/presentation/formsy/FormComponent.ts

```diff
@@ -0,0 +1,130 @@
+// https://github.com/christianalfoni/formsy-react/issues/191#issuecomment-144872142
+
+import { Mixin, ValidationErrors } from 'formsy-react';
+import { omit } from 'lodash';
+import * as React from 'react';
+
+/*
+ * Here the Formsy.Mixin is wrapped as an abstract class, so that it can
+ * be used effectively with TypeScript.
+ */
+export interface IFormComponentProps {
+  name?: string;
+  value?: string | string[] | number;
+  validations?: string;
+  validationError?: string;
+  validationErrors?: ValidationErrors;
+  required?: boolean;
+  _validate?: Function;
+}
+
+export interface IFormComponentState {
+  _value: any;
+  _isRequired: boolean;
+  _isValid: boolean;
+  _isPristine: boolean;
+  _pristineValue: any,
+  _validationError: string;
+  _externalError: string;
+  _formSubmitted: boolean;
+}
+
+/** A base class for Formsy form components */
+export abstract class FormComponent<VALUE, PROPS extends IFormComponentProps, STATE extends IFormComponentState> extends React.Component<PROPS, STATE> {
+  public static propnames: (keyof IFormComponentProps)[] = ['name', 'value', 'validations', 'validationError', 'validationErrors', 'required', '_validate'];
+
+  public static defaultProps: IFormComponentProps = {
+    name: null,  // this can be set to whatever, since it will be overwritten when child components are created
+    validationError: '',
+    validationErrors: {}
+  };
+
+  public static otherProps(props: any): any {
+    return omit(props, FormComponent.propnames);
+  }
+
+  constructor(props: PROPS, context: any) {
```

Comment:
Yeah, sorry... It's a convention I've used in the past to indicate that the symbol is a generic type parameter.  Instead of just using `P` and `S` like I've seen elsewhere.  

LMK